api/desecapi/models.py

```diff
@@ -195,209 +168,66 @@ def pdns_id(self):
 
         return name
 
-    # This method does not use @transaction.atomic as this could lead to
-    # orphaned zones on pdns.
-    def create_on_pdns(self):
-        """
-        Create zone on pdns
+    @property
+    def normalized_name(self):
+        return self.name.rstrip('.') + '.'
 
-        This method should only be called for new domains when they are created,
-        or when the domain was created with a locked account and not yet propagated.
-        """
+    @property
+    def parent_domain_name(self):
+        return self.normalized_name.split('.', 1)[1].rstrip('.')
```

Comment:
1.) This throws `IndexError` if the domain is a TLD.
2.) Can't we just `return self.name.split('.', 1)[1]`?
3.) All functions that use `parent_domain_name` also need the other part. Maybe replace this method by
```
def partition_name(self):
    subname, _, parent_name = self.name.partition('.')
    return subname, parent_name or None
```
Consumers then can use `subname, parent_name = self.partition_name()` and also use `parent_name is None` to check for TLD.